### PR TITLE
cargo-show-asm: init at 0.1.24

### PR DESCRIPTION
**Pull request by oxalica**
###### Description of changes

https://github.com/pacak/cargo-show-asm

`cargo-show-asm` is a reimpl for unmaintained `cargo-asm`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-show-asm/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    rev = version;
+    hash = "sha256-ahkKUtg5M88qddzEwYxPecDtBofGfPVxKuYKgmsbWYc=";
+  };
+
+  cargoHash = "sha256-S7OpHNjiTfQg7aPmHEx6Q/OV5QA9pB29F3MTIeiLAXg=";
+
+  nativeBuildInputs = [ pkg-config ];
+  buildInputs = [ openssl ];
+
+  postInstall = ''
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-show-asm/default.nix:18]: please use `installShellCompletion` from `installShellFiles`

File: pkgs/development/tools/rust/cargo-show-asm/test-add.nix [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Test the basic usage of cargo-asm on x86_64.
```
Comment (figsoda) [inline pkgs/development/tools/rust/cargo-show-asm/test-add.nix:1]: perhaps making a directory here would be more readable?

Comment (oxalica) [reply pkgs/development/tools/rust/cargo-show-asm/test-add.nix:1]: You mean a sub-directory for the test? Kind of overkill. Maybe renaming it to `test-basic-x86_64.nix` should be fine.

Comment (figsoda) [reply pkgs/development/tools/rust/cargo-show-asm/test-add.nix:1]: > You mean a sub-directory for the test?

yeah

i don't really mind the name, we can keep it as it is

**Comment by figsoda** [comment]
darwin build is still failing, maybe we can just mark it broken if you can't figure something out and worry about it later

**Comment by oxalica** [comment]
> darwin build is still failing, maybe we can just mark it broken if you can't figure something out and worry about it later

Ok, I give up with darwin build.

**Review by figsoda — APPROVED** [review]

**Pull request merged**